src/main/java/me/jellysquid/mods/sodium/client/render/pipeline/BlockRenderer.java

```diff
@@ -138,6 +138,8 @@ private void renderQuad(BlockRenderView world, BlockState state, BlockPos pos, M
             int lm = light.lm[srcIndex];
 
             sink.writeQuad(x, y, z, color, u, v, lm);
+
+            buffers.getRenderData().addSprite(src.getSprite());
```

Comment:
This doesn't need to be done for every vertex - you should move this outside the loop. You should also check if the sprite is null first - see this code from the stable branch: https://github.com/CaffeineMC/sodium-fabric/blob/1.16.x/stable/src/main/java/me/jellysquid/mods/sodium/client/render/chunk/compile/ChunkMeshBuilder.java#L108

Reply:
I will try to find a better solution